### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by rabernat**
@nathanieltarshish: this supersedes your recent PR (#22)

I also refactored the structure of `generators.py` and the corresponding tests significantly.

We now can pass test FloatSets with land masks to the test functions.

### CODE REVIEW

File: floater/generators.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def xyz_to_geo(xyz_coord):
 class FloatSet(object):
     """Represents a set of initial float positions on a regular grid."""
 
-    def __init__(self, xlim, ylim, dx=1., dy=1.):
+    def __init__(self, xlim, ylim, dx=1., dy=1., model_grid=None):
```
Comment (rabernat) [inline floater/generators.py]: We now pass `model_grid` as an optional keyword argument when initializing the FloatSet object.

File: floater/generators.py [hunk 2]
```diff
@@ -70,55 +79,11 @@ def get_rectmesh(self):
             2D array of float y coordinates
         """
 
-        return np.meshgrid(self.x, self.y)
-
-    def get_oceancoords(self, model_grid, mesh='rect'):
-        """Get the coordinates of float positions taking into account the land mask.
```
Comment (rabernat) [inline floater/generators.py]: This function has been eliminated.

File: floater/generators.py [hunk 3]
```diff
@@ -70,55 +79,11 @@ def get_rectmesh(self):
...
-        -------
-        floats_ocean: np.ndarray
-            1D array of float coordinate subarrays:
-            e.g. floats_ocean[i] = [some_float_lon, some_float_lat] 
-        """
-
-        xx, yy = self.get_rectmesh()
+        xx, yy = np.meshgrid(self.x, self.y)
+        if self.model_grid is not None:
+            xx, yy = _subset_floats_from_mask(xx, yy, self.model_grid)
```
Comment (rabernat) [inline floater/generators.py]: when outputting a mesh, we check for the presence of a model grid and, if necessary, apply the land mask by calling ``_subset_floats_from_mask``, a standalone function.

File: floater/generators.py (hunk 1 above)
Comment (nathanieltarshish) [inline floater/generators.py]: Great!

File: floater/generators.py [hunk 4]
```diff
@@ -40,6 +40,14 @@ def __init__(self, xlim, ylim, dx=1., dy=1.):
             grid spacing in x direction
         dy : float
             grid spacing in y direction
+    model_grid : dictionary
+        the following key value pairs are expected
+            'land_mask': np.ndarray of bools
+                2d array of dimensions in C order: shape==(len(lat), len(lon))
```
Comment (rabernat) [inline floater/generators.py]: I also modified the dimension order of land_mask.

In python, the convention is to use C-order ([row-major order](https://en.wikipedia.org/wiki/Row-_and_column-major_order)) for multidimensional arrays.

You will have to change your user code accordingly.

File: floater/generators.py (hunk 3 above)
Comment (nathanieltarshish) [inline floater/generators.py]: This is much cleaner.

File: floater/generators.py (hunk 4 above)
Comment (nathanieltarshish) [inline floater/generators.py]: Okay, will do.

**Pull request merged**